cirq/google/optimizers/optimize_for_sycamore.py

```diff
@@ -12,73 +12,113 @@
-    optimizers.EjectZ(tolerance=_TOLERANCE).optimize_circuit,
-    optimizers.DropNegligible(tolerance=_TOLERANCE).optimize_circuit,
-]  # type: List[Callable[[circuits.Circuit], None]]
+
+def _get_xmon_optimizers(tolerance: float, tabulation: Optional[GateTabulation]) \
+        -> List[Callable[['cirq.Circuit'], None]]:
+    if tabulation is not None:
+        raise ValueError("Gate tabulation not supported for xmon")
+
+    return [
+        convert_to_xmon_gates.ConvertToXmonGates().optimize_circuit,
+        optimizers.MergeInteractions(tolerance=tolerance,
+                                     allow_partial_czs=False).optimize_circuit,
+        lambda c: optimizers.merge_single_qubit_gates_into_phased_x_z(
+            c, tolerance),
+        optimizers.EjectPhasedPaulis(tolerance=tolerance).optimize_circuit,
+        optimizers.EjectZ(tolerance=tolerance).optimize_circuit,
+        optimizers.DropNegligible(tolerance=tolerance).optimize_circuit,
+    ]
+
+
+def _get_xmon_optimizers_part_cz(tolerance: float, tabulation: Optional[GateTabulation]) \
+        -> List[Callable[['cirq.Circuit'], None]]:
+    if tabulation is not None:
+        raise ValueError("Gate tabulation not supported for xmon")
+    return [
+        convert_to_xmon_gates.ConvertToXmonGates().optimize_circuit,
+        optimizers.MergeInteractions(tolerance=tolerance,
+                                     allow_partial_czs=True).optimize_circuit,
+        lambda c: optimizers.merge_single_qubit_gates_into_phased_x_z(
+            c, tolerance),
+        optimizers.EjectPhasedPaulis(tolerance=tolerance).optimize_circuit,
+        optimizers.EjectZ(tolerance=tolerance).optimize_circuit,
+        optimizers.DropNegligible(tolerance=tolerance).optimize_circuit,
+    ]
+
+
+def _get_sycamore_optimizers(tolerance: float, tabulation: Optional[GateTabulation]) \
+        -> List[Callable[['cirq.Circuit'], None]]:
+    return [
+        ConvertToSycamoreGates(tabulation=tabulation).optimize_circuit,
+        lambda c: optimizers.merge_single_qubit_gates_into_phased_x_z(
+            c, tolerance),
+        optimizers.EjectPhasedPaulis(tolerance=tolerance).optimize_circuit,
+        optimizers.EjectZ(tolerance=tolerance).optimize_circuit,
+        optimizers.DropNegligible(tolerance=tolerance).optimize_circuit,
+    ]
+
+
+def _get_sqrt_iswap_optimizers(tolerance: float, tabulation: Optional[GateTabulation]) \
+        -> List[Callable[['cirq.Circuit'], None]]:
+    if tabulation is not None:
+        raise ValueError("Gate tabulation not supported for sqrt_iswap")
+    return [
+        ConvertToSqrtIswapGates().optimize_circuit,
+        lambda c: optimizers.merge_single_qubit_gates_into_phased_x_z(
+            c, tolerance),
+        optimizers.EjectPhasedPaulis(tolerance=tolerance).optimize_circuit,
+        optimizers.EjectZ(tolerance=tolerance).optimize_circuit,
+        optimizers.DropNegligible(tolerance=tolerance).optimize_circuit,
```

Comment:
I wonder if we should combine these last 3 optimizers into a common function since all 4 classes use them.

Reply:
done